controllers/users/users.js

```diff
@@ -26,105 +26,72 @@ const sendVerificationEmail = user => {
     `
   }
 
-  transporter.sendMail(mailOptions, (err, data) => {
+  transporter.sendMail(mailOptions, err => {
     if(err) throw err;
   })
 }
 
-exports.activateUser = async (req, res) => {
+exports.activateUser = (req, res) => {
   const { activationKey } = req.params;
-  await User.findOneAndUpdate({ activationKey }, { emailVerified: true })
-  .then(() => {
-    res.status(200).json({ msg: messages.USER_ACTIVATED });
-  })
-  .catch(() => {
-    res.status(400).json({ msg: messages.SERVER_ERROR });
-  });
+  User.findOneAndUpdate({ activationKey }, { emailVerified: true })
+    .then(() => {
+      res.status(200).json({ msg: messages.USER_ACTIVATED });
+    })
+    .catch(() => {
+      res.status(400).json({ msg: messages.SERVER_ERROR });
+    });
 }
 
 exports.registerUser = (req, res) => {
   const errors = validationResult(req);
 
   if (!errors.isEmpty()) {
-    return res
-      .status(400)
-      .json({ errors: errors.array({ onlyFirstError: true }) });
+    return res.status(400).json({ errors: errors.array({ onlyFirstError: true }) });
   }
 
-  const { firstName, lastName, email, password } = req.body;
+  const { firstName, lastName, email = email.toLowerCase(), password } = req.body;
 
-  User.findOne({ email: email.toLowerCase() }).then(async (user) => {
+  User.findOne({ email }).then(user => {
     if (user) {
-      return res
-        .status(400)
-        .json({ errors: [{ msg: messages.EMAIL_ALREADY_EXISTS }] });
+      return res.status(400).json({ errors: [{ msg: messages.EMAIL_ALREADY_EXISTS }] });
     } else {
-      const newUser = new User({
-        firstName,
-        lastName,
-        email: email.toLowerCase(),
-        password
+      const newUser = new User({ 
+        firstName, 
+        lastName, 
+        email, 
+        password: bcrypt.hashSync(password), 
+        activationKey: bcrypt.hashSync(email).replace(/\//g, '') 
```

Comment:
To better hashing we can add salt to this like: 
bcrypt.hashSync(password, 10)
same for activationKey aswell

Reply:
hashSync has two arguments 
1. string to hash
2. salt (default is 10)